core/exception/source_encoding_name_spec.rb

```diff
@@ -1,9 +1,23 @@
 require_relative '../../spec_helper'
 
 describe "Encoding::UndefinedConversionError#source_encoding_name" do
-  it "needs to be reviewed for spec completeness"
+  it "returns the source encoding name" do
+    ec = Encoding::Converter.new("ISO-8859-1", "EUC-JP")
+    begin
+      ec.convert("\xa0")
+    rescue Encoding::UndefinedConversionError => e
+      e.source_encoding_name.should == "UTF-8"
+    end
+  end
 end
 
 describe "Encoding::InvalidByteSequenceError#source_encoding_name" do
-  it "needs to be reviewed for spec completeness"
+  it "returns the source encoding name" do
+    ec = Encoding::Converter.new("EUC-JP", "ISO-8859-1")
+    begin
+      ec.convert("\xa0")
+    rescue Encoding::InvalidByteSequenceError => e
+      e.source_encoding_name.should == "EUC-JP"
+    end
+  end
```

Comment:
Note that normally the `raise_error` matcher is preferred to a direct `begin/rescue/end` as in:
```ruby
    ec = Encoding::Converter.new("ISO-8859-1", "EUC-JP")
    -> {
      ec.convert("\xa0")
    }.should raise_error(Encoding::UndefinedConversionError) { |e|
      e.destination_encoding.should == Encoding::EUC_JP
    }
```
Notably this fails if there is no exception thrown with a helpful message.
However, for such a simple cases with `should` only in the `rescue`, this would also fail as each spec much have at least one `should`.

Reply:
I added documentation for this form in 8f06f95b3c2b6c2e30f88dd1668f6c7c3fe89c38